x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/textstructure/structurefinder/TextStructure.java

```diff
@@ -213,6 +224,11 @@ public TextStructure(StreamInput in) throws IOException {
         quote = in.readBoolean() ? (char) in.readVInt() : null;
         shouldTrimFields = in.readOptionalBoolean();
         grokPattern = in.readOptionalString();
+        if (in.getVersion().onOrAfter(Version.V_8_5_0)) {
+            ecsCompatibility = getNonNullEcsCompatibilityString(in.readOptionalString());
+        } else {
+            ecsCompatibility = null;
```

Comment:
Setting it to `null` here is a loophole that spoils the idea of always setting it to a valid string in the other places:

```suggestion
            ecsCompatibility = getNonNullEcsCompatibilityString(null);
```